src/components/SetPassword.js

```diff
@@ -0,0 +1,120 @@
+        this.submit = this.submit.bind(this);
+    }
+
+    change(e) {
+        this.setState({
+            [e.target.name]: e.target.value
+        })
+    }
+
+    showMessage = (msg) => {
+        this.setState({
+            isMessageShown: true,
+            message: msg
+        })
+    }
+
+    hideMessage = () => {
+        if (this.state.isMessageShown) {
+            this.setState({
+                isMessageShown: false,
+                message: ''
+            })
+        }
+    }
+
+    setMessageColor = (color) => {
+        this.setState({messageColor: color})
+    }
+
+    submit(e) {
+        e.preventDefault();
+        if (comparePasswords(this.state.password1, this.state.password2) && isPasswordStrong(this.state.password1)) {
+                    const url = REACT_APP_SET_PASS_PATH;
+                    axios.post(url,
+            {
+                "Email": getEmailFromPath(window.location.href),
+                "Password": this.state.password1,
+                "ConfirmPassword": this.state.password2,
+                "Token": getTokenFromPath(window.location.href)
+            },
+            {
+                headers: {
+                    'Cache-Control': 'no-cache',
+                    'Content-Type': 'application/json',
+                    "Access-Control-Allow-Origin": "*"
+                }
+            })
+            .then(resp => {
+                console.log(resp);
+                if (resp.data.isSuccess) {
+                    this.setMessageColor("#134a1e")
+                    this.showMessage(changedPasswordText());
+                } else {
+                    this.setMessageColor("maroon")
+                    this.showMessage(expiredTokenText());
+                }
+            })
+            .catch(error => {
+                console.log(error.resp);
+            });
+        } else {
+            this.setMessageColor("maroon")
+            this.showMessage(invalidPasswordsText());
+        }
+
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <div className="auth-area">
+                    <div className="form-div">
+                        <h1 id="title-login">Set a new password</h1>
+                        <h3 className="subtitle">{getEmailFromPath(window.location.href)}</h3>
+                        <form name="recover-form" onSubmit={e => this.submit(e)}>
+                            <div className="field-area">
+                                <input type="password" name="password1" id="email-input" placeholder="New password"
+                                       onChange={e => {
+                                           this.change(e)
+                                       }} onClick={this.hideMessage}/>
+                            </div>
+                            <div className="field-area">
+                                <input type="password" name="password2" id="email-input" placeholder="Confirm password"
```

Comment:
Why this input has the same id, like above? The id should be unique per website.

Reply:
I agree with Łukasz, the ids should be unique to prevent situation that we base our business logic on ids and each id does not point only one html tag. @AdamSarach, please correct it in the next commit :)